scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env node
+const path = require('path');
+const fs = require('fs');
+const execSync = require('child_process').execSync;
+const got = require('got');
+const { isArray } = require('@salesforce/ts-types');
+
+const circleciBaseUrl = 'https://circleci.com/api/v2/';
+
+/**
+ * Script to locate and launch NUTs across sfdx modules
+ *
+ * Script requires that the env var CIRCLECI_API_TOKEN is set to a valid CircleCi API Token
+ *
+ * args:
+ *   sfdx-cli version: z.y.z | latest | latest-rc
+ *
+ * Flow:
+ *   load sfdx-cli package.json
+ *   For each entry in oclif.plugins
+ *     Identify repo organization and project i.e. @salesforce/plugin-auth -> salesforcecli/plugin-auth via "npm show @salesforce/plugin-auth homepage"
+ *     Identify if plugin supports script test:nuts i.e. "npm show @salesforce/plugin-auth scripts | grep test:nuts"
+ *     Check if org/project is known in circleci
+ *       Trigger test nuts workflow for plugin with required parameters
+ */
+
+const manualRepoOverrides = {
+  'salesforce-alm': {
+    org: 'salesforce-cli',
+    repo: 'toolbelt',
+  },
+};
+
+const hasTestNuts = (module) => {
+  const results = execSync(`npm show ${module.name}@${module.version} scripts`);
+  return results.includes('test:nuts');
+};
+
+const getNpmForModule = (timeCreated, module, sections) => {
```

Comment:
@amphro I modified this function with the goal of either returning a single version of a module or undefined. sfdx-cli has a reference to plugin-auth@^1.3.0 and since that allows version selection for all version in the 1.x lineage, the plugin was always being selected as having `test:nuts`. This change takes some additional information, timeCreated, which is the timestamp assigned to the sfdx-cli version. If the npm show returns more than one version (as it does for auth) the code will sort the returned version is version timestamp descending and then locate the version of a plugin whose version timestamp is <= sfdx-cli timestamp.

This also has the benefit of fixing an exact version for any plugin selected, which will be used by NUTS to checkout the correct version of the plugin that was used at the time sfdx-cli was published.

Reply:
Can you add a comment explaining this. Took me a second to wrap my head around this.

Also, I don't think **ANY** core plugin should be carroted, including plugin-auth. We should probably keep this code in here in case it does happen, we can handle it, but I think we should add it to the pinned dep list.